lifelines/utils/__init__.py

```diff
@@ -896,6 +894,9 @@ def check_complete_separation(df, events, durations):
 
 
 def check_nans_or_infs(df_or_array):
+    if hasattr(df_or_array, "dtype") and df_or_array.dtype.kind not in "biuf":
+        raise ValueError("Values must be a subtype of number: so no strings, timedeltas, datetimes, objects, etc.")
+
```

Comment:
Prior to this change it was possible to use CoxPHFitter with an event information in a column of dtype object.
This was useful when calling CoxPHFitter as part of a pipeline where pre-processing steps included missing data. By using the object dtype, missing data could be represented and dropped before calling CoxPHFitter. After this change, instead the column type needs to change within the pipeline.

As a workaround, it seems that dtype float can be used throughout the pipeline.

Maybe check_nans_or_infs could be changed to accept columns of type object if all rows have acceptable content?

Reply:
hey @ManuelaS, happy to make changes that work with your workflow. However, can you describe the following in more detail?

> By using the object dtype, missing data could be represented and dropped before calling CoxPHFitter. After this change, instead the column type needs to change within the pipeline.

Wouldn't CoxPHFitter just blow up anyways if missing data was present?

Reply:
In my case, I have a sklearn pipeline with multiple steps ending with a cox model (wrapped with sklearn-lifelines). The early steps of the pipeline run on data that may have missing survival information. These rows are always dropped before CoxPHFitter is run, but with the new version dropping the rows is not sufficient - the dtype of the column also need to change.

For example: 
```
pipeline = sklearn.pipeline.Pipeline(
    steps=[
        ('consensus_cluster', py_utils.cluster_transformer.ClusterTransformer(
            py_utils.consensus_clustering.ConsensusClustering(
                n_clusters=3, base_estimator=sklearn.cluster.AgglomerativeClustering(), n_estimators=1000),
            sklearn.ensemble.RandomForestClassifier(n_estimators=1000, random_state=0))),
        ('cox', py_utils.drop_missing_estimator.DropMissingEstimator(
            sklearn_lifelines.estimators_wrappers.CoxPHFitterModel(
                duration_column='overall_survival_months', event_col='has_event')))
    ],
)
```

In this case DropMissingEstimator drops any rows with missing data in them, but is otherwise not specific to CoxPHFitterModel and does not know what the different columns in y are or what dtype they should be.

It is of course not to difficult to make this work with the new lifelines code, but I just wanted to mention it in case this was an unintended consequence of the change.

Reply:
I think I understand: your column has some missing data, and `DropMissingEstimator` will get rid of those, but the column dtype might still be `object` which will fail at my lifelines LOC. 

<img width="657" alt="screen shot 2019-02-16 at 5 44 31 pm" src="https://user-images.githubusercontent.com/884032/52906056-8f826b80-3212-11e9-8dbc-6c4104555ceb.png">


Yea, this is a legit problem. I'll fix in 0.19.0 to handle this problem.

Reply:
Yes, that's exactly it. Thank you very much!

Reply:
Try the just released 0.19.0 - that should work again. Note there are some important other API changes to be aware of too: https://lifelines.readthedocs.io/en/latest/Changelog.html#api-changes

Reply:
Thanks!